src/client/Microsoft.Identity.Client/Instance/Discovery/NetworkMetadataProvider.cs

```diff
@@ -73,7 +77,7 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryResponse instanceDi
           Uri authority,
           RequestContext requestContext)
         {
-            var client = new OAuth2Client(requestContext.Logger, _httpManager, _telemetryManager);
+            var client = new OAuth2Client(requestContext.Logger, _httpManager, _telemetryManager, _deviceAuthManager);
```

Comment:
I don't think this is necessary. You can just pass null here, i.e. the device auth manager should be null since you don't need to perform the extra complexity of device auth for instance discovery.